extensions/wikia/Email/EmailController.class.php

```diff
@@ -189,22 +194,45 @@ protected function getSubject() {
 	 * @return string
 	 */
 	protected function getBody() {
-		$moduleBody = $this->app->renderView(
-			get_class( $this ),
-			'body',
-			$this->request->getParams()
-		);
+		$css = file_get_contents( __DIR__ . '/styles/main.css' );
 
 		$html = $this->app->renderView(
 			get_class( $this ),
 			'main',
 			[
-				'content' => $moduleBody,
+				'content' => $this->getContent(),
 				'footerMessages' => $this->getFooterMessages()
-
 			]
 		);
 
+		$html = $this->inlineStyles( $html, $css );
+
+		return $html;
+	}
+
+	/**
+	 * Renders the content unique to each email. Should be overridden in child classes.
+	 */
+	protected function getContent() {
```

Comment:
I tried changing `getSubject` to an abstract method in my branch and it worked as expected, you can still call it in the base class even though it's the child class that has to implement it.

How about we make `getSubject` and `getContent` abstract methods and make EmailController an abstract class?